### PR TITLE
Enumerable

### PR DESCRIPTION
**Pull request by Said-laasri**
1. Created a class MyList that has an instance variable @list.
In MyList implement a method #each that yields successive members of @list and uses the MyEnumerable module.

2. Created a module MyEnumerable that implements the following methods (they should have the same functionality as methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html)):

- #all?
- #any?
- #filter

# Create our list
irb> list = MyList.new(1, 2, 3, 4)
=> #<MyList: @list=[1, 2, 3, 4]>
# Test #all?
irb> list.all? {|e| e < 5}
=> true
irb> list.all? {|e| e > 5}
=> false
# Test #any?
irb> list.any? {|e| e == 2}
=> true
irb> list.any? {|e| e == 5}
=> false

# Test #filter
irb> list.filter {|e| e.even?}
=> [2, 4]

### CODE REVIEW

File: class.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require_relative 'module'
+class Mylist
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+end
```
Comment (GzimAsani) [inline class.rb]: - [ ] `MyList` class needs to have a #each method that yields all members of @list in succession.
image

File: README.md [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # DECODING MORSE-CODE
 
-In this Project we are asked to decode a string with MORSE-CODE using Ruby Programming language.
-We wrote several functions to perform these tasks.
+In this project we are asked to create an enumerable object from scratch in Ruby.
```
Comment (GzimAsani) [inline README.md]: - [x] Add instructions on how to clone and run/test your app

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Said-laasri and @ogagaoghene          :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your enumerables they work as expected :+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Said-laasri @ogagaoghene,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class.rb (hunk 1 above)
Comment (Said-laasri) [reply class.rb]: @GzimAsani Thank you for reviewing my project, I will make the necessary corrections as suggested

File: README.md (hunk 2 above)
Comment (Said-laasri) [reply README.md]: @GzimAsani Thank you for reviewing my project, I will make the necessary corrections as suggested

**Comment by Said-laasri** [comment]
@DammyShittu Thank you for reviewing my project

**Pull request merged**